nmigen/hdl/mem.py

```diff
@@ -92,6 +118,37 @@ def __getitem__(self, index):
 
 
 class ReadPort(Elaboratable):
+    """A memory read port.
+
+    Parameters
+    ----------
+    memory : :class:`Memory`
+        Memory associated with the port.
+    domain : str
+        Clock domain. Defaults to ``"sync"``. If set to ``"comb"``, the port is asynchronous.
+        Otherwise, reads have a latency of 1 clock cycle.
```

Comment:
I would rephrase this along the lines of "the read data becomes available on the next clock cycle", since otherwise there's a bit of ambiguity with pipelined reads where that'd be an *extra* clock cycle.